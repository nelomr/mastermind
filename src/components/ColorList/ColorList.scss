.color-list {
  display: flex;
  flex: 0 0 40px;
  width: 40px;

  @include breakpoint($min-desktop) {
    flex-direction: column;
  }
}