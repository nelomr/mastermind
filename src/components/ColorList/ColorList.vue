<template>
  <div class="color-list" :class="{'is-loaded': loaded }">
    <ColorButton 
      v-for="(color, index) in listColors" 
      :key="'color'+index"
      @clicked="onClickButton"
      :color="color"
      :index="index"
      :class="{'is-selected': indexSelected == index }"
    />
  </div>
</template>
<script>
import EventBus from '@/modules/event-bus';
import ColorButton from '@/components/ColorButton/ColorButton';
import { mapState } from 'vuex';

export default {
    name: 'ColorList',
    components: {
      ColorButton
    },
    data() {
      return {
        indexSelected: null
      }
    },
    created() {
      EventBus.$on('reset', () => {
        this.reset();
      });
    },
    computed: {
      ...mapState([
        'listColors',
        'loaded'
      ])
    },
    methods: {
      onClickButton(value) {
        this.indexSelected = value;
      },
      reset() {
        Object.assign(this.$data,this.$options.data.call(this));
      }
    }
}
</script>
<style lang="scss">
  @import "./ColorList.scss";
</style>