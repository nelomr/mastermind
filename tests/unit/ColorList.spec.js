import Vue from 'vue';
import Vuex from 'vuex';
import { shallowMount, createLocalVue } from '@vue/test-utils';
import ColorList from '@/components/ColorList/ColorList.vue';

const localVue = createLocalVue();
localVue.use(Vuex);

describe('ColorList.vue', () => {
    let listColors = null;
    let store;

    let computed = {};
  
    beforeEach(() => {
      computed = {
        listColors() {
          return listColors;
        },
      }
      store = new Vuex.Store({
      });
    });

    const build = () => {
        const wrapper = shallowMount(ColorList, {
            computed,
            mocks: {$store: store}
        });

        return {
            wrapper
        };
    };

  it('renders the component', () => {
    const {wrapper} = build();

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders color buttons list ', () => {
    const build = () => {
      const wrapper = shallowMount(ColorList,
        {
          computed: {
            listColors() {
              return ['red','blue','green','yellow','orange','white'];
            }
          },
          mocks: {$store: store}
        }
      );

      return {
          wrapper
      };
    };
    const {wrapper} = build();
    
    expect(wrapper.html()).toMatchSnapshot();
  });
  
})
