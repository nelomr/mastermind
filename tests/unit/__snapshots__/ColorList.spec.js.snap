// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorList.vue renders color buttons list  1`] = `
<div class="color-list">
  <colorbutton-stub color="red" index="0" indexrow="0" class=""></colorbutton-stub>
  <colorbutton-stub color="blue" index="1" indexrow="0" class=""></colorbutton-stub>
  <colorbutton-stub color="green" index="2" indexrow="0" class=""></colorbutton-stub>
  <colorbutton-stub color="yellow" index="3" indexrow="0" class=""></colorbutton-stub>
  <colorbutton-stub color="orange" index="4" indexrow="0" class=""></colorbutton-stub>
  <colorbutton-stub color="white" index="5" indexrow="0" class=""></colorbutton-stub>
</div>
`;

exports[`ColorList.vue renders the component 1`] = `<div class="color-list"></div>`;
